iOS/second-hand/second-hand/MyAccount/Controller/NotLoginMyAccountViewController.swift

```diff
@@ -0,0 +1,171 @@
+        }
+    }
+    
+    internal func loginStatus() {
+        if isLogin {
+            setLoginedUI()
+        }
+        else {
+            setNotLoginedUI()
+        }
+    }
+    
+    private func setLoginedUI() {
+        for subview in view.subviews {
+            subview.removeFromSuperview()
+        }
+        self.view.addSubview(loginViewController.view)
+        loginViewController.didMove(toParent: self)
+        loginViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            loginViewController.view.topAnchor.constraint(equalTo: view.topAnchor),
+            loginViewController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            loginViewController.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            loginViewController.view.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+        ])
+    }
+    
+    private func setNotLoginedUI() {
+        for subview in view.subviews {
+            subview.removeFromSuperview()
+        }
+        setNavigationBar()
+        setLoginButton()
+        setJoinButton()
+        setLoginedConstraints()
+        setGithubLoginButton()
+        contour.backgroundColor = UIColor.neutralBorder
+    }
+    
+    
+    private func setNavigationBar() {
+        self.navigationItem.title = "내 계정"
+    }
+    
+    private func setLoginButton() {
+        loginButton.frame = CGRect(x: 0, y: 0, width: 361, height: 52)
+        loginButton.setTitle("로그인", for: .normal)
+        loginButton.titleLabel?.font = UIFont.subHead
+        loginButton.backgroundColor = .orange
+        loginButton.layer.cornerRadius = 8
+        loginButton.layer.masksToBounds = true
+        loginButton.addTarget(self, action: #selector(loginButtonTouched), for: .touchUpInside)
+    }
+    
+    private func setGithubLoginButton() {
+        githubLoginButton.frame = CGRect(x: 0, y: 0, width: 361, height: 52)
+        githubLoginButton.setTitle("GitHub 계정으로 로그인", for: .normal)
+        githubLoginButton.titleLabel?.font = UIFont.subHead
+        githubLoginButton.backgroundColor = .black
+        githubLoginButton.layer.cornerRadius = 8
+        githubLoginButton.layer.masksToBounds = true
+        githubLoginButton.addTarget(self, action: #selector(githubLoginButtonTouched), for: .touchUpInside)
+    }
+    
+    @objc func joinButtonTouched() {
+        present(UINavigationController(rootViewController: joinViewController), animated: true)
+    }
+    
+    @objc private func loginButtonTouched() { //일단 로그인 성공했다고 가정
+        let loginNotification = Notification(name: NSNotification.Name("LOGIN"))
+        NotificationCenter.default.post(name: loginNotification.name, object: nil, userInfo: nil)
+        // TODO: 싱글톤으로 전역처럼 사용할 변수만들어야 할 듯
+        isLogin = true
+        
+        //얘를 사용하려니까 자식뷰컨트롤러라 사용이 안됨
+        //present(loginViewController, animated: true)
+        //대체한것이 이 메서드 호출
+        setLoginedUI()
```

Comment:
자식 뷰컨트롤러라 사용이 안되는 이유가 무엇일까요?
대체하여 사용법을 찾는것도 중요하지만, `왜`안되는지 찾아보는것도 중요한 학습일 수 있어요. 🙂
[ChildViewController](https://medium.com/geekculture/child-view-controllers-lifecycle-part-6-a1b2828ffe26)
```suggestion
        view.addSubview(loginViewController.view)
```

Reply:
하하.. 한 화면을 구성하기위해 부모컨트롤러와 자식 컨트롤러가 있는건데 자식 컨트롤러로 present를 하려고 하다니..  😂😂😂 자식컨트롤러는 화면 구성을 도와주기위해 화면을 분해하여 세팅을 잡아주는 컨트롤러라서 전환이 안되는걸로 이해했습니다 !